extensions/amp-accordion/1.0/accordion.type.js

```diff
@@ -56,23 +57,31 @@ AccordionDef.AccordionHeaderProps;
 /**
  * @typedef {{
  *   as: (string|PreactDef.FunctionalComponent|undefined),
+ *   role: (string|undefined)
  *   className: (string|undefined),
+ *   id: (string|undefined),
  *   children: (?PreactDef.Renderable|undefined),
  * }}
  */
 AccordionDef.AccordionContentProps;
 
 /**
  * @typedef {{
- *   role: (string|undefined),
+ *   id: (string),
```

Comment:
`id` and `role` here and for `ContentProps` allow `undefined`

Reply:
Right, let's chat about this in real time.  I did do some re-factoring of the type definitions in the restructuring PR https://github.com/ampproject/amphtml/pull/31083.  So 

- HeaderProps
- ContentProps

refer to the Header and ContentShims passed from amp side and for these, many of the props will never be undefined and the entire set of props are not public

- AccordionSectionProps
- AccordionHeaderProps
- AccordionContentProps

Refer to the actually public Accordion Components defined in `accordion.js`

Reply:
Refactored names in latest revision.  Summary here: https://github.com/ampproject/amphtml/pull/31136#issuecomment-728285011